### DISCUSSION TITLE
get_psets() does not extract all the properties

### DISCUSSION THREAD

**Discussion by Borna07** [IfcOpenShell/IfcOpenShell#1828] [Q&A]
I noticed a discrepancy between the properties visible in an ifc viewer like solibri or bimcollab zoom and the properties that the "get_pset" function extracts. 

Does anyone know why this happens, or rather how do I get all the properties?

These are screenshots of the same IfcCovering element

![Properties_BIMcollab](https://user-images.githubusercontent.com/63143991/138713222-366ea1e2-60d8-4584-8830-d188d47064f1.PNG)
![Properties_IfcOpenShell](https://user-images.githubusercontent.com/63143991/138713225-2950e9da-7895-4e49-9a95-092933438f2a.png)

**Comment by aothms** [comment]
Some viewers calculate / augment properties and quantities on the fly. Did you check an exact element you miss in the ifcopenshell output and search whether it exists in the file?

**Reply by arunkvijay** [reply]
@aothms I have a similiar issue in an ifc file. any pointers on how do i generate these missing quantities using ifcopenshell python. for ex: I have the Length quantity missing for 1 of the `IfcColumn`

**Reply by Moult** [reply]
See ifcopenshell.util.shape https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/util/shape.py - you can calculate all sorts of quantities.

**Comment by Borna07** [comment]
I was aware that some viewers calculate the properties on the fly, but they always appeared under the viewer native property tab and not under a property tab defined in the ifc scheme like here. That is what confuses me here...

I am not sure how I would search for the exact missing element with the ifcopenshell. 

I tried manually searching the ifc file for the missing terms (Fußboden - Estrich, Fußboden - Parkett Kastanie...) and found them under different ifc elemenets like IFCSHAPEASPECT and IFCMATERIALCONSTITUENT but not under a IFCQUANTITY or IFCPROPERTYSINGLEVALUE where I would expect to find it...

**Reply by vulevukusej** [reply]
@Borna07 I´ve experienced the same thing with other software as well.  I think what @aothms says is what you confirmed yourself by searching for the terms directly in the ifc file.  I´ve also had other bizarre scenarios where non-existent psets are created in the viewer which don't exist in the ifc file.  Sometimes it's nice that these viewers are able to 'interpret' the file in such a way that is beneficial to your specific use-case, but it's not technically correct and can be a nightmare when you have multiple stakeholders using different viewers, all wondering why they see something different! ;)
